gateway-ha/src/main/java/io/trino/gateway/ha/security/LbFormAuthManager.java

```diff
@@ -48,7 +49,10 @@ public LbFormAuthManager(FormAuthConfiguration configuration,
             Map<String, String> pagePermissions)
     {
         this.presetUsers = presetUsers;
-        this.pagePermissions = pagePermissions;
+        this.pagePermissions = new HashMap<>();
+        if (pagePermissions != null) {
```

Comment:
Is this really nullable in product code?

Reply:
I noticed that it is not nullable in production. (value though could be null so ignored it)
Tested with giving below conf.
```
pagePermissions:
  admin:
  user: dashboard_history
  api:
```

Reply:
should we rather change to test code so that it doesn't send null? or have sth like below?
```java
this.pagePermissions = Objects.requireNonNullElse(pagePermissions, Map.<String, String>of()).entrySet().stream()
            .filter(entry -> entry.getValue() != null)  // Filter out entries with null values
            .collect(toImmutableMap(
                entry -> entry.getKey().toUpperCase(),
                Map.Entry::getValue
            ));
```

Reply:
We should change the test in this case.

Reply:
didn't know that ide was giving a warnging about  `@NotNull` annotations too. i changed it to new HashMap 
![image](https://github.com/trinodb/trino-gateway/assets/25147023/5f71dc07-e99b-48e8-ad41-385135eb2dcc)

Reply:
> i changed it to new HashMap

Please avoid mutable map.

Reply:
i don't quite understand when you mean by `adding test with null values` if pagePermission cannot be null in anycase. what is the purpose of having null value??
ps. changed to use immutable map in test case

Reply:
I mean covering the following line:
```java
.filter(entry -> entry.getValue() != null)  // Filter out entries with null values
```

Reply:
if i add a test for map with null values, it fails to build with following error
```java
    @Test
    public void testNullValuesInPagePermission()
        throws Exception
    {
        Map<String, UserConfiguration> presetUsers = ImmutableMap.of(
                "user1", new UserConfiguration("admin, user, api", "pass1"));
        Map<String, String> pagePermission = ImmutableMap.of(
                "admin", "",
                "user", null,
                "api", "");
        LbFormAuthManager authentication = new LbFormAuthManager(null, presetUsers, pagePermission);
        assertThat(authentication.authenticate(new BasicCredentials("user1", "pass1")))
                .isTrue();
    }
```
```
/Users/user/work/trino-gateway-chaho/gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbAuthenticator.java:144:25
java: [NullArgumentForNonNullParameter] Null is not permitted for this parameter.
    (see https://errorprone.info/bugpattern/NullArgumentForNonNullParameter)
```

Reply:
Please use `unmodifiableMap` with `HashMap`.

Reply:
fixed using unmodifiableMap & hashmap. 
- test code checks for null values for either key/value for page permission.